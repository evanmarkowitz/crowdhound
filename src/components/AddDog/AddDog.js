import React, { Component } from 'react';
import './AddDog.css';

export class AddDog extends Component {

  //  loadImg = (e) => {
  //   e.preventDefault();
  //   if (chooseFile.files[0]){
  //     reader.readAsDataURL(chooseFile.files[0]);
  //     reader.onload = makeImg;
  //   }
  // }

  render() {
    return (
      <section className="add-dog">
        <form className="add-dog-form">
          <h4 className="add-dog-title">Add your dog</h4>
          <label for="dog-name-input" className="label">Name</label>
          <input id="dog-name-input" className="input" type="text" />
          <label for="dog-breed-input" className="label">Breed</label>
          <input id="dog-breed-input" className="input" type="text" />
          <label for="dog-birthdate-input" className="label">Birthdate</label>
          <input id="dog-birthdate-input" className="input" type="date" />
          <label for="dog-weight-input" className="label">Weight</label>
          <input id="dog-weight-input" className="input" type="text" />
          <label for="dog-description-input" className="label">Description</label>
          <input id="dog-description-input" className="input" type="text" />
          <label for="dog-activity-level-input" className="label">Activity Level</label>
          <select id="dog-activity-level-input" className="input">
            <option vlaue="1">Low</option>
            <option vlaue="2">Medium</option>
            <option vlaue="3">High</option>
          </select>
          <label for="dog-photo-input" className="label">Photo</label>
          <input id="dog-photo-input" className="input" type="file" />
          <input type="button" className="input" value="Add" id="add-dog-btn" />
        </form>
      </section>
    )
  }
}

export default AddDog;
