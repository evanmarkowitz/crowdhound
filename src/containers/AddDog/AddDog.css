.add-dog {
  padding-top: 60px;
  height: calc(100vh - 60px);
}

.add-dog-form {
  margin-top: 20px;
  margin-left: auto;
  margin-right: auto;
  width: 302px;
  display: flex;
  flex-direction: column;
}

.add-dog-title {
  color: #f08080;
  font-size: 36px;
  text-align: center;
  margin-bottom: 25px;
}

.label {
  color: rgb(15, 141, 170);
  padding-right: 5px;
  font-size: 22px;
  margin-top: 30px;
}

.input {
  outline: none;
  width: 300px;
  height: 30px;
  border-radius: 5px;
  font-size: 16px;
  border: 1px solid rgb(29, 187, 223);
  padding-left: 7px;
}

#add-dog-btn {
  margin-top: 40px;
  cursor: pointer;
  background: rgba(29, 188, 223, 0.253);
}

#dog-description-input {
  resize: none;
  height: 200px;
  padding: 15px;
}

#add-dog-btn:hover {
  background: rgb(29, 187, 223);
  color: #fff;
}

#add-dog-btn:active {
  background: rgb(21, 164, 196);
  color: #fff;
}