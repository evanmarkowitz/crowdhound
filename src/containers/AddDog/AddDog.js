import React, { Component } from 'react';
import './AddDog.css';
import { addDogQuery, createPhoto } from '../../api/apiCallsNew'
import ReactFileReader from 'react-file-reader';
import { connect } from 'react-redux';
import {Link } from 'react-router-dom'




export class AddDog extends Component {

  constructor() {
    super()
    this.state = {
      name: '',
      breed: '',
      birthdate: '',
      weight: 0,
      longDesc: '',
      activityLevel: 0,
      photo: '', 
      popUp: false
    }
  }

  handleChange = (e) => {
    this.setState({[e.target.name]: e.target.value})
  }

  handleFiles = (files) => {
    this.setState({photo: files.base64})
  }

  sendDog = async () => {
    const {name, breed, birthdate, weight, activityLevel } = this.state
    let opts = {
      method: 'POST',
      headers: { "Content-Type": "application/json"},
      body: JSON.stringify({query: addDogQuery(
        name, activityLevel, breed, weight, birthdate,
      )})
    }
    try{
      let response = await fetch(`https://staging-crowdhound-be.herokuapp.com/graphql?token=${this.props.token}`, opts)
      await console.log(response)
      let parsedResponse = await response.json()
      await console.log(parsedResponse)
      return parsedResponse.data.createDog.dog.id
  
    } catch(error) {
      console.log(error)
    }

  }

  sendPhoto = async (id, file) => {
    let query = createPhoto('Dog',  id, 'Dog Photo')
    let photoOpts = {
      method: 'POST',
      headers: { "Content-Type": "application/json"},
      body: JSON.stringify({
        file: file
      })
    }
    try {
      await fetch(`https://staging-crowdhound-be.herokuapp.com/graphql?token=${this.props.token}&query=${query}`, photoOpts)
    } catch(error) {
      await console.log(error)
    }
  }

  submitDog = async () => {
    try {
      let dogId = await this.sendDog()
      await console.log(dogId)
      // let addPhoto = await this.sendPhoto(dogId, this.state.photo)
      // let parsedResponse = addPhoto.json()
      // console.log(parsedResponse)
    } catch(error){
      console.log(error)
    }
    
  }


  render() {
    return (
      <section className="add-dog">
        <form className="add-dog-form">
          <h4 className="add-dog-title">Add your dog</h4>
          <label for="dog-name-input" className="label">Name</label>
          <input id="dog-name-input" className="input" type="text" name="name" onChange={this.handleChange}/>
          <label for="dog-breed-input" className="label">Breed</label>
          <input id="dog-breed-input" className="input" type="text" name="breed" onChange={this.handleChange}/>
          <label for="dog-birthdate-input" className="label">Birthdate</label>
          <input id="dog-birthdate-input" className="input" type="date" name="birthdate" onChange={this.handleChange}/>
          <label for="dog-weight-input" className="label">Weight (lbs)</label>
          <input id="dog-weight-input" className="input" type="text" name="weight" onChange={this.handleChange}/>
          <label for="dog-description-input" className="label">Description</label>
          <textarea id="dog-description-input" className="input"  type="text" name="longDesc" onChange={this.handleChange}></textarea>
          <label for="dog-activity-level-input" className="label" >Activity Level</label>
          <select id="dog-activity-level-input" className="input" name="activityLevel" onChange={this.handleChange}>
            <option value={0}>Low</option>
            <option value={1}>Medium</option>
            <option value={2}>High</option>
          </select>
          <label for="dog-photo-input" className="label">Photo</label>
          <ReactFileReader handleFiles={this.handleFiles} base64={true} multipleFiles={false}>
            <button id="dog-photo-input" className="input" type="button">Add photo</button>
          </ReactFileReader>
          <Link to={`/userprofile/${this.props.currentId}`}>
            <input type="button" className="input" value="Add" id="add-dog-btn"  onClick={this.submitDog}/>
          </Link>
        </form>
      </section>
    )
  }
}

export const mapStateToProps = state => ({
  token: state.currentUser.token,
  currentId: state.currentUser.id
})


export default connect(mapStateToProps)(AddDog)
