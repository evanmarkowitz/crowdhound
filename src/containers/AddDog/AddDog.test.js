import { shallow } from 'enzyme';
import { AddDog } from './AddDog'
import React from 'react';


describe('AddDog', () => {
  let wrapper;

  beforeEach(() => {

    wrapper = shallow(<AddDog />)
  })
  
  it('should match the snapshot', () =>
    expect(wrapper).toMatchSnapshot()
  )


})