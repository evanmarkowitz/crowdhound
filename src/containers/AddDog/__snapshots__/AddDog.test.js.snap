// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddDog should match the snapshot 1`] = `ShallowWrapper {}`;
